gapis/resolve/metrics.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resolve
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/messages"
+	"github.com/google/gapid/gapis/service"
+	"github.com/google/gapid/gapis/service/path"
+)
+
+func Metrics(ctx context.Context, p *path.Metrics) (*service.Metrics, error) {
+	switch p.Type {
+	case path.Metrics_MEMORY:
+		return memoryMetrics(ctx, p)
+	default:
+		return nil, fmt.Errorf("Metrics type %v not implemented", p.Type)
+	}
+
+}
+
+// FIXME DO NOT MERGE: Not sure where the best spot for this is, it can't go in gapis/api because api can't depend on service
```

Comment:
This is likely not the best spot for this interface, maybe it should go in `gapis/service` along with the MemoryLayout proto definition?

Reply:
It most likely belongs in `gapis/api`. This is similar to the Mesh API: `gapis/api/mesh.go`.

Reply:
Ok, and then the Metrics proto definition should be moved to `gapis/api/service.proto`?